migration-engine/connectors/sql-migration-connector/src/sql_schema_calculator.rs

```diff
@@ -213,7 +213,7 @@ impl<'a> SqlSchemaCalculator<'a> {
                             },
                         );
 
-                        let field = model.fields().find(|f| &f.db_name() == column_name).unwrap();
+                        let field = model.fields().find(|f| &f.name() == &dml_field.name).unwrap();
```

Comment:
This line was the main problem. This calls `single_database_name` underneath. This function crashes on compound relation fields.

Reply:
Indeed - maybe I should try to write tests for this, the migration engine should be ready for when the `ARR` part of `AUMFIDARR` part needs to be tested. I think there are already a few tests, but it apparently doesn't cover enough.

Reply:
That would be amazing. 👍